code/__HELPERS/maths.dm

```diff
@@ -326,3 +326,61 @@ proc/arctan(x)
 	assert_eq(count_set_bitflags(65536|32768), 2)
 	assert_eq(count_set_bitflags(1|4|16), 3)
 #endif
+
+// Basic geometry things.
+
+/vector/
+	var/x = 0
+	var/y = 0
+
+/vector/New(var/x, var/y)
+	src.x = x
+	src.y = y
+
+/vector/proc/duplicate()
+	return new /vector(x, y)
+
+/vector/proc/euclidian_norm()
+	return sqrt(x*x + y*y)
+
+/vector/proc/squared_norm()
+	return x*x + y*y
+
+/vector/proc/noramlise()
```

Comment:
normalize?

Reply:
It's the british spelling
Convert the vector into a vector with norm 1, I can put it as a comment if needed

Reply:
I think he was talking about it being _'noramlise'_ in the example, rather than _'normalise'._

Reply:
:face_with_head_bandage: i'll fix it